listener.go

```diff
@@ -100,6 +100,12 @@ func initListener(network, addr string, options *Options) (l *listener, err erro
 		sockOpt := socket.Option{SetSockOpt: socket.SetSendBuffer, Opt: options.SocketSendBuffer}
 		sockOpts = append(sockOpts, sockOpt)
 	}
+	if strings.HasPrefix(network, "udp") {
+		if sockoptFn := socket.SetMulticastMembership(network, addr); sockoptFn != nil {
```

Comment:
I'm not sure about this, is it a good idea to enable this by default?

Reply:
`SetMulticastMembership` will check if `addr` is a multicast address and returns early if it is not. 
Anyway, I refactored the code to make it clear.

Reply:
LGTM.